src/components/Filter/components/FilterBase/FilterBase.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { forwardRef, ReactNode } from 'react';
+
+import useTheme from '../../../../hooks/useTheme';
+import { useTypeColorToColorMatch } from '../../../../hooks/useTypeColorToColorMatch';
+import { generateTestDataId } from '../../../../utils/helpers';
+import Icon from '../../../Icon';
+import { AcceptedIconNames } from '../../../Icon/types';
+import {
+  buttonSpanStyle,
+  buttonStyle,
+  buttonWrapperStyle,
+  childrenWrapperStyle,
+  dividedButtonStyle,
+  divider,
+  labelSpanStyle,
+  valueSpanStyle,
+  wrapperStyle,
+} from '../../Filter.style';
+import { FilterType, StyleType } from '../../types';
+
+// eslint-disable-next-line react/display-name
+export const FilterBase = forwardRef<
+  HTMLButtonElement,
+  {
+    dataTestId?: string;
+    children?: ReactNode;
+    disabled: boolean;
+    handleOpen: () => void;
+    onClear: () => void;
+    selectedItemLabel?: string;
+    open: boolean;
+    hasSelectedValue: boolean;
+    label?: string;
+    iconName: AcceptedIconNames;
+    iconColor: string;
+    iconSize?: number;
```

Comment:
why we need this ? dont we have same sizes for icons ? if no why (can we ask the design team)?